File: app/src/main/java/com/designpatterns/demo/behavioral/state/java/RunningState.java
package com.designpatterns.demo.behavioral.state.java;

/**
 * 电梯在当前状态下能做的事情
 *
 * @author zhangzhiguo
 * @version 1.0.0
 * @email: 810903987@qq.com
 * @date 2018/6/1 上午11:39
 * @project_name DesignPatternsDemo
 */
public class RunningState extends LiftState {
    @Override
    public void open() {

    }

    @Override
    public void close() {

    }

    @Override
    public void run() {
        System.out.println("电梯正在运行");

    }

    @Override
    public void stop() {
//        设置电梯状态
        super.mContext.setLiftState(Context.stoppingState);
        super.mContext.getLiftState().stop();

    }
}

File: app/src/main/java/com/designpatterns/demo/behavioral/state/kotlin/RunningState.kt
package com.designpatterns.demo.behavioral.state.kotlin

/**
 * 电梯在当前状态下能做的事情
 *
 * @author  zhangzhiguo
 * @email: 810903987@qq.com
 * @date 2018/6/1 下午2:08
 * @project_name DesignPatternsDemo
 * @version 1.0.0
 *
 */
class RunningState:LiftState(){
    override fun open() {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun close() {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun run()= println("电梯运行状态")

    override fun stop() {
        super.mContext.setLiftState(Context.stoppingState)
        super.mContext.getLiftState().stop()
    }

}